code/game/machinery/computer/warrant.dm

```diff
@@ -0,0 +1,220 @@
+			<tr>
+			<th>Crime</th>
+			<th>Fine</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			<th>Amount Due</th>
+			<th>Make Payment</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["citation"])
+				var/owed = c.fine - c.paid
+				dat += "<tr><td>[c.crimeName]</td>"
+				dat += "<td>$[c.fine]</td>"
+				dat += "<td>[c.author]</td>"
+				dat += "<td>[c.time]</td>"
+				if(owed > 0)
+					dat += "<td>$[owed]</td>"
+					dat += "<td><A href='?src=[REF(src)];choice=Pay;field=citation_pay;cdataid=[c.dataId]'>\[Pay\]</A></td>"
+				else
+					dat += "<td colspan='2'>All Paid Off</td>"
+				dat += "</tr>"
+			dat += "</table>"
+
+			dat += "<br>Minor Crimes:"
+			dat +={"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+			<tr>
+			<th>Crime</th>
+			<th>Details</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["mi_crim"])
+				dat += "<tr><td>[c.crimeName]</td>"
+				dat += "<td>[c.crimeDetails]</td>"
+				dat += "<td>[c.author]</td>"
+				dat += "<td>[c.time]</td>"
+				dat += "</tr>"
+			dat += "</table>"
+
+			dat += "<br>Major Crimes:"
+			dat +={"<table style="text-align:center;" border="1" cellspacing="0" width="100%">
+			<tr>
+			<th>Crime</th>
+			<th>Details</th>
+			<th>Author</th>
+			<th>Time Added</th>
+			</tr>"}
+			for(var/datum/data/crime/c in current.fields["ma_crim"])
+				dat += "<tr><td>[c.crimeName]</td>"
+				dat += "<td>[c.crimeDetails]</td>"
+				dat += "<td>[c.author]</td>"
+				dat += "<td>[c.time]</td>"
+				dat += "</tr>"
+			dat += "</table>"
+		else
+			dat += {"<span>** No security record found for this ID **</span>"}
+
+	dat = dat.Join()
+	var/datum/browser/popup = new(user, "warrant", "Security Warrant Console", 600, 400)
+	popup.set_content(dat)
+	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+
+/obj/machinery/computer/warrant/Topic(href, href_list)
+	if(..())
+		return
+
+	switch(href_list["choice"])
+		if("Login")
+			eject_id(usr)
+			if(!scan)
+				var/obj/item/card/id/O = usr.is_holding_item_of_type(/obj/item/card/id)
+				if(O)
+					if(!usr.transferItemToLoc(O, src))
+						return
+					scan = O
+				else
+					to_chat(usr, "<span class='danger'>No valid ID.</span>")
+
+		if("Logout")
+			eject_id(usr)
+			current = null
+
+		if("Pay")
+			for(var/datum/data/crime/p in current.fields["citation"])
+				if(p.dataId == text2num(href_list["cdataid"]))
+					var/datum/bank_account/R = scan.registered_account
+					to_chat(world, "Found your citation and your account!")
+					var/diff = p.fine - p.paid
+					var/pay = FLOOR(input(usr, "Please enter how much you would like to pay:", "Citation Payment", 50) as num, 1)
+					if(!pay || pay < 0)
+						to_chat(usr, "<span class='warning'>You're pretty sure that's not how money works.</span>")
+						return
+					if(pay > diff)
+						to_chat(usr, "<span class='notice'>You only owe $[diff] credit\s to pay off this fine.</span>")
```

Comment:
Should just cap it at that and pay really.(Or allow paying above for sec bribes)

Reply:
Done and done!